v-next/hardhat-verify/src/internal/sourcify.ts

```diff
@@ -0,0 +1,376 @@
+import type {
+  SourcifyErrorResponse,
+  SourcifyLookupResponse,
+  SourcifyVerificationStatusResponse,
+  SourcifyVerificationResponse,
+} from "./sourcify.types.js";
+import type {
+  VerificationProvider,
+  VerificationStatusResponse,
+} from "./types.js";
+import type {
+  Dispatcher,
+  DispatcherOptions,
+  HttpResponse,
+} from "@nomicfoundation/hardhat-utils/request";
+import type { VerificationProvidersConfig } from "hardhat/types/config";
+import type { CompilerInput } from "hardhat/types/solidity";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+import { ensureError } from "@nomicfoundation/hardhat-utils/error";
+import { sleep } from "@nomicfoundation/hardhat-utils/lang";
+import {
+  getProxyUrl,
+  getRequest,
+  postJsonRequest,
+  shouldUseProxy,
+} from "@nomicfoundation/hardhat-utils/request";
+
+export const SOURCIFY_PROVIDER_NAME: keyof VerificationProvidersConfig =
+  "sourcify";
+
+const VERIFICATION_STATUS_POLLING_SECONDS = 3;
+
+export const SOURCIFY_REPO_URL = "https://repo.sourcify.dev";
+export const SOURCIFY_API_URL = "https://sourcify.dev/server";
```

Comment:
Are the staging and local urls commonly used? How often do users use a different repo or api url?